### PR TITLE
feat(mocha): add missing failZero option

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://mochajs.org/api/mocha#failZero

### CODE REVIEW

File: types/mocha/index.d.ts [hunk 1]
```diff
@@ -180,6 +180,13 @@ declare class Mocha {
      */
     delay(): boolean;
 
+    /**
+     * Fails test run if no tests encountered with exit-code 1.
+     *
+     * @see https://mochajs.org/api/mocha#failZero
+     */
+    failZero(): boolean;
```
Comment (gabritto) [inline types/mocha/index.d.ts:188]: According to https://mochajs.org/api/mocha#failZero, this function should take an optional boolean parameter, no? And the return type should be `this`:

```suggestion
    failZero(enabled?: boolean): this;
```

**Review by gabritto — CHANGES_REQUESTED** [review]

File: types/mocha/index.d.ts (hunk 1 above)
Comment (gabritto) [reply types/mocha/index.d.ts:188]: It's up to you, you don't have to fix the existing ones if you don't want, but that would be nice.

Comment (johanneswuerbach) [reply types/mocha/index.d.ts]: Only fixed the current method for now, I'll see if I've time to change the others in a follow up PR.

**Review by gabritto — APPROVED** [review]

**Comment by johanneswuerbach** [comment]
Ready to merge

**Pull request merged**